### PR TITLE
Update license.go (handled error)

### PR DESCRIPTION
**Pull request by TheInvincibleRalph**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/28401

```release-note
NONE
```

### CODE REVIEW

File: server/channels/api4/license.go [hunk 1]
```diff
@@ -93,7 +95,10 @@ func addLicense(c *Context, w http.ResponseWriter, r *http.Request) {
 	defer file.Close()
 
 	buf := bytes.NewBuffer(nil)
-	io.Copy(buf, file)
+	if _, err := io.Copy(buf, file); err != nil {
+		c.Err = model.NewAppError("addLicense", "api.license.add_license.copy.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
```
Comment (hanzei) [inline server/channels/api4/license.go:99]: This is a new error ID right? Please run `make i18n-extract` to update the translation file and add the error text.

File: server/channels/api4/license.go [hunk 2]
```diff
@@ -239,24 +244,32 @@ func requestTrialLicense(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getPrevTrialLicense(c *Context, w http.ResponseWriter, r *http.Request) {
+	// Check if the LicenseManager is initialized
```
Comment (hanzei) [inline server/channels/api4/license.go:247]: While the comment is not wrong, it also unrelated to the purpose of the PR. Can you please remove it and the other comments you added?

File: server/channels/api4/license.go [hunk 3]
```diff
@@ -239,24 +244,32 @@ func requestTrialLicense(c *Context, w http.ResponseWriter, r *http.Request) {
...
 		clientLicense = utils.GetClientLicense(license)
 	} else {
 		clientLicense = utils.GetSanitizedClientLicense(utils.GetClientLicense(license))
 	}
 
-	w.Write([]byte(model.MapToJSON(clientLicense)))
+	w.Header().Set("Content-Type", "application/json")
+	if _, err := w.Write([]byte(model.MapToJSON(clientLicense))); err != nil {
+		c.Logger.Warn("Error while writing response", mlog.Err(err))
+		http.Error(w, "Failed to write response", http.StatusInternalServerError)
```
Comment (hanzei) [inline server/channels/api4/license.go:273]: If `w.Write` fails, there is no point in sending another message to the client. Let's log the error and be done.

**Review by hanzei — CHANGES_REQUESTED** [review]
Please also remote this file from the list in the golangci-lint config file.

File: server/channels/api4/license.go (hunk 3 above)
Comment (TheInvincibleRalph) [reply server/channels/api4/license.go:273]: Done. Thank you.

File: server/channels/api4/license.go (hunk 2 above)
Comment (TheInvincibleRalph) [reply server/channels/api4/license.go:247]: Done. Thank you!

File: server/channels/api4/license.go (hunk 1 above)
Comment (TheInvincibleRalph) [reply server/channels/api4/license.go:99]: Done already. Thank you for the review.

**Review by hanzei — CHANGES_REQUESTED** [review]
Can you please update `.golangci.yml`? Thanks!

**Comment by TheInvincibleRalph** [comment]
> Can you please update `.golangci.yml`? Thanks!

Done, @hanzei. Thank you.

**Review by hanzei — APPROVED** [review]
Awesome : 🚀

**Comment by TheInvincibleRalph** [comment]
> Awesome : 🚀

Thank you, @hanzei!

File: server/i18n/en.json
Comment (hanzei) [inline server/i18n/en.json:1]: The ordering seems wrong. Can you please run `make i18n-extract` again?

Comment (TheInvincibleRalph) [reply server/i18n/en.json:1]: Roger

**Comment by TheInvincibleRalph** [comment]
Change effected, @hanzei.

**Comment by hanzei** [comment]
@nickmisasi Gentle reminder to review the PR :wink:

**Review by nickmisasi — APPROVED** [review]
LGTM. Thanks @TheInvincibleRalph

**Pull request merged**